utils/ort/src/main/kotlin/CopyrightStatementsProcessor.kt

```diff
@@ -26,32 +26,6 @@ import java.util.SortedSet
 
 import org.ossreviewtoolkit.utils.common.collapseWhitespace
```

Comment:
It would be really interesting to know this use case. Would you mind providing an example how this use case requires the copyrights to be processed / merged?

Reply:
@porsche-rbieniek can provide you the details about the case for using `determineParts` in report generation.

But besides that, to all @oss-review-toolkit/kotlin-devs can we now *please* proceed and stop blocking this unless there are good reasons beyond questions about the scope or those asked just out of curiosity?

Reply:
I would like to understand the use case before approving it, if it is possible to share it.

Reply:
The Porsche use case is that we need to 'group' copyright statements by the copyright holder. 

We need to understand two facts here:
- list of copyright holders per package
- list of copyright statements issued by a particular copyright holder per package

To give an example copyright statement list here:
- "(c) Zeno Rocha",
- "Copyright (C) The Bootstrap Authors",
- "Copyright (c) 2011-2021 The Bootstrap Authors",
- "Copyright (c) 2011-2021 Twitter, Inc.",
- "Copyright (c) 2021 Bryan Braun",
- "Copyright 2011 $ year pkg.author",
- "Copyright 2011-2021 The Bootstrap Authors",
- "Copyright 2011-2021 The Bootstrap Authors (https://github.com/twbs/bootstrap/graphs/contributors)",
- "Copyright 2011-2021 Twitter, Inc.",
- "copyright 2011-2021 the Bootstrap Authors (https://github.com/twbs/bootstrap/graphs/contributors) and Twitter, Inc. (https://twitter.com)"

I would need to group as this:
- "The Bootstrap Authors": { "Copyright (C) The Bootstrap Authors", "Copyright (c) 2011-2021 The Bootstrap Authors", "Copyright 2011-2021 The Bootstrap Authors (https://github.com/twbs/bootstrap/graphs/contributors)", "copyright 2011-2021 the Bootstrap Authors (https://github.com/twbs/bootstrap/graphs/contributors) and Twitter, Inc. (https://twitter.com)" }
- "Twitter, Inc.": { "Copyright (c) 2011-2021 Twitter, Inc.", "copyright 2011-2021 the Bootstrap Authors (https://github.com/twbs/bootstrap/graphs/contributors) and Twitter, Inc. (https://twitter.com)" }
- "Zeno Rocha": { "(c) Zeno Rocha" }
- "Bryan Braun": { "Copyright (c) 2021 Bryan Braun" }
- "$ year pkg.author": {"Copyright 2011 $ year pkg.author" }

Reply:
Thanks @rbieniek !

Can you explain how the logic is supposed to arrive at these groups? 
In particular, do you think you will have a heuristic which produces the output you want solely based on the given statements above, or would there be additional (manual human) input needed?

Reply:
> do you think you will have a heuristic which produces the output you want solely based on the given statements above

I honestly don't see how the outcome of this conversation is in anyway helpful, let alone required, for being able to review the code changes in this PR. But @rbieniek / @porsche-rbieniek can you please yet respond timely to move this forward?

Reply:
Speaking from the perspective of Porsche. it would be helpful to have some kind heuristic, either built-in or implemented seperately, for grouping the copyright statements by the copyright holder for easing the life of the (human) analysts who need to review the output of ORT anyway. And who need to take appropiate actions based on that outcome.

Speaking from my personal point of view, as an user of ORT I simply asked for some capability already built into ORT to be exposed in a way that I can make use of it in custom code I implement for supporting internal workflows. 
I do not fully understand why and how the discussion about relatively small code change could have gotten so much out of the way into a somewhat fundamental discussion.

Reply:
You ask me to review a series of refactoring changes, without providing the use case / context.
Now I've just asked for the information which I would like to consider when thinking about the proposed approach.
I have really have no clue what problem you have with that.